src/io/std_file_descriptor.cr

```diff
@@ -0,0 +1,64 @@
+require "../signal"
+require "c/time"
+
+class IO::StdFileDescriptor < IO::FileDescriptor
+  @read_timer_id : LibC::TimerT
+  @write_timer_id : LibC::TimerT
+  @blocking = true
+
+  def initialize(fd, blocking = false)
+    ret = LibC.timer_create(LibC::CLOCK_MONOTONIC, nil, out @read_timer_id)
+    raise Errno.new("timer_create") if ret == -1
```

Comment:
Since STDIN, STDOUT and STDERR can only be read from or written to, I guess we could use a single timer.

Reply:
Well, darwin and openbsd only provide `setitimer`, where we can't have different timers, so I'm now using only one.